### PR TITLE
Added language filter for the category meta box

### LINKED ISSUE #38: Categories metabox
- Investigate if we can modify the terms retrieved by this metabox.
See if the metabox uses the `get_terms` function (we have a `get_terms` filter) or if we can use `pre_get_terms`.

The focus is filter the terms by language meta field as we do with post types.

If we can't modify the metabox we'll need remove it and create a clone with those filters.

### PR DESCRIPTION
**Pull request by chandonn-zz**
Fixes #38

### CODE REVIEW

File: admin/class-monk-admin.php [hunk 1]
```diff
@@ -298,6 +298,30 @@ public function monk_save_post_meta_box( $post_id ) {
...
 	/**
+	 * Function to filter the query inside the category meta box using the post language
+	 *
+	 * @param    $term_query instance of WP_Term_Query class 
+	 *
+	 * @since    1.0.0
+	*/
+	public function monk_apply_terms_filter( $term_query ) {
+		$screen = get_current_screen();
+		if ( 'edit' === $screen->parent_base ) {
```
Comment (brenoalvs) [inline admin/class-monk-admin.php]: Add conditional to check if is category query
Add conditional to check `$lang` query var

File: admin/class-monk-admin.php [hunk 2]
```diff
@@ -298,6 +298,30 @@ public function monk_save_post_meta_box( $post_id ) {
...
+			$post_id   = get_the_id();
+			$meta_lang = sanitize_title( get_post_meta( $post_id, '_monk_post_language', true ) );
+			$term_args = array(
```
Comment (brenoalvs) [inline admin/class-monk-admin.php]: Remove default args

File: admin/class-monk-admin.php [hunk 3]
```diff
@@ -298,6 +298,30 @@ public function monk_save_post_meta_box( $post_id ) {
...
+				'hide_empty' => false,
+				'fields'     => 'all',
+				'count'      => true,
+				'meta_key'   => 'monk-language',
```
Comment (brenoalvs) [inline admin/class-monk-admin.php]: Use `_monk_term_language` meta key

File: admin/class-monk-admin.php [hunk 4]
```diff
@@ -231,7 +231,7 @@ public function monk_post_meta_box_field_render( $post ) {
 		
 		wp_nonce_field( basename( __FILE__ ), 'monk_post_meta_box_nonce' );
 
-		if ( $post_default_language == '' ) {
+		if ( '' == $post_default_language ) {
```
Comment (brenoalvs) [inline admin/class-monk-admin.php]: Use `===`

File: admin/class-monk-admin.php [hunk 5]
```diff
@@ -298,6 +298,30 @@ public function monk_save_post_meta_box( $post_id ) {
 	}
 
 	/**
+	 * Function to filter the query inside the category meta box using the post language
+	 *
+	 * @param    $term_query instance of WP_Term_Query class 
+	 *
+	 * @since    1.0.0
+	*/
+	public function monk_apply_terms_filter( $term_query ) {
```
Comment (brenoalvs) [inline admin/class-monk-admin.php]: Modify function name to `monk_category_language_filter`

**Review by brenoalvs — CHANGES_REQUESTED** [review]
@chandonn take a look

File: admin/class-monk-admin.php (hunk 4 above)
Comment (chandonn-zz) [reply admin/class-monk-admin.php]: Fixed

File: admin/class-monk-admin.php (hunk 5 above)
Comment (chandonn-zz) [reply admin/class-monk-admin.php]: Fixed

File: admin/class-monk-admin.php (hunk 3 above)
Comment (chandonn-zz) [reply admin/class-monk-admin.php]: Fixed

File: admin/class-monk-admin.php (hunk 2 above)
Comment (chandonn-zz) [reply admin/class-monk-admin.php]: Fixed

File: admin/class-monk-admin.php (hunk 1 above)
Comment (chandonn-zz) [reply admin/class-monk-admin.php]: Fixed

File: admin/class-monk-admin.php [hunk 6]
```diff
@@ -298,6 +298,37 @@ public function monk_save_post_meta_box( $post_id ) {
...
+	*/
+	public function monk_category_language_filter( $term_query ) {
+		$screen = get_current_screen();
+		if ( 'edit' === $screen->parent_base && 'category' === $term_query->query_vars['taxonomy'] ) {
+			$term_args  = '';
+			$post_id    = get_the_id();
+			$meta_lang  = sanitize_title( get_post_meta( $post_id, '_monk_post_language', true ) );
+			$query_lang = $_GET['lang'];
+			
+			if ( isset( $meta_lang ) && '' !== $meta_lang ) {
```
Comment (brenoalvs) [inline admin/class-monk-admin.php]: Use `||`

File: admin/class-monk-admin.php [hunk 7]
```diff
@@ -298,6 +298,37 @@ public function monk_save_post_meta_box( $post_id ) {
...
+	 *
+	 * @since    1.0.0
+	*/
+	public function monk_category_language_filter( $term_query ) {
+		$screen = get_current_screen();
+		if ( 'edit' === $screen->parent_base && 'category' === $term_query->query_vars['taxonomy'] ) {
+			$term_args  = '';
+			$post_id    = get_the_id();
+			$meta_lang  = sanitize_title( get_post_meta( $post_id, '_monk_post_language', true ) );
+			$query_lang = $_GET['lang'];
```
Comment (brenoalvs) [inline admin/class-monk-admin.php]: Add sanitize function

**Review by brenoalvs — CHANGES_REQUESTED** [review]
@chandonn

**Pull request merged**